framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -501,52 +501,38 @@ public Receiver visit(ClassExpr expr, FlowExpressionContext context) {
+                        return new ClassName(types.getPrimitiveType(TypeKind.BYTE));
                     case SHORT:
-                        return new ClassName(
-                                TypesUtils.typeFromClass(
-                                        short.class, types, env.getElementUtils()));
+                        return new ClassName(types.getPrimitiveType(TypeKind.SHORT));
                     case INT:
-                        return new ClassName(
-                                TypesUtils.typeFromClass(int.class, types, env.getElementUtils()));
+                        return new ClassName(types.getPrimitiveType(TypeKind.INT));
                     case CHAR:
-                        return new ClassName(
-                                TypesUtils.typeFromClass(char.class, types, env.getElementUtils()));
+                        return new ClassName(types.getPrimitiveType(TypeKind.CHAR));
                     case FLOAT:
-                        return new ClassName(
-                                TypesUtils.typeFromClass(
-                                        float.class, types, env.getElementUtils()));
+                        return new ClassName(types.getPrimitiveType(TypeKind.FLOAT));
                     case LONG:
-                        return new ClassName(
-                                TypesUtils.typeFromClass(long.class, types, env.getElementUtils()));
+                        return new ClassName(types.getPrimitiveType(TypeKind.LONG));
                     case DOUBLE:
-                        return new ClassName(
-                                TypesUtils.typeFromClass(
-                                        double.class, types, env.getElementUtils()));
+                        return new ClassName(types.getPrimitiveType(TypeKind.DOUBLE));
                 }
             } else if (expr.getType().isVoidType()) {
-                return new ClassName(
-                        TypesUtils.typeFromClass(void.class, types, env.getElementUtils()));
+                return new ClassName(types.getNoType(TypeKind.VOID));
             }
 
             throw new ParseRuntimeException(
                     constructParserException(expr.toString(), "is an unparsable class literal"));
         }
 
+        /** @param expr an array creation expression, with dimensions and/or initializers. */
         @Override
         public Receiver visit(ArrayCreationExpr expr, FlowExpressionContext context) {
 
             // Parse the type as a class literal
             Receiver receiver =
-                    visit(
-                            StaticJavaParser.parseExpression(expr.getElementType() + ".class")
-                                    .asClassExpr(),
-                            context);
+                    StaticJavaParser.parseExpression(expr.getElementType() + ".class")
+                            .accept(this, context);
```

Comment:
What I tried to do here is call `visit(ClassExpr, FlowExpressionContext)` directly, because the `String` would be parsed s a `ClassExpr` anyways. If I don't append `.class` to the end, if `int[]` is parsed, then it would choose the `NameExpr` visitor for `int` (which is `expr.getElementType().asString()`, which throws an exception. Is there a way to improve this?